nexus/src/app/background/tasks/region_replacement.rs

```diff
@@ -170,8 +170,79 @@ impl BackgroundTask for RegionReplacementDetector {
             };
 
             for request in requests {
+                // If the replacement request is in the `requested` state and
+                // the request's volume was soft-deleted or hard-deleted, avoid
+                // sending the start request and instead transition the request
+                // to completed
+
+                let volume_deleted = match self
```

Comment:
Unfortunately no - this queries the database, and the result could change immediately after the query.

If it does change to deleted in the middle of the saga, then that could be a problem, but both `volume_replace_region` and `volume_replace_snapshot` check for if the volume was hard-deleted during the transaction................. and a38b751bc updates this to also check if it is soft-deleted too!

Reply:
I just want to be sure that we can handle a delete if it can happen, really anywhere during this saga.  If we can handle that, either by failing the replacement or handling it at the end, then I'm good :)

Reply:
Yeah, the saga will unwind. There's more work to do in a related follow up commit though, I found a case where the start saga will do some extra unnecessary work allocating regions if there's a hard delete of the volume in the middle of its execution.